contracts/src/v0.8/functions/dev/1_0_0/FunctionsCoordinator.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
+import {IFunctionsBilling, FunctionsBilling, Route, ITypeAndVersion} from "./FunctionsBilling.sol";
+import {OCR2Base} from "./ocr/OCR2Base.sol";
+import {Functions} from "./Functions.sol";
+import {IOwnable} from "../../../shared/interfaces/IOwnable.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsSubscriptions} from "./interfaces/IFunctionsSubscriptions.sol";
+
+/**
+ * @title Functions Coordinator contract
+ * @notice Contract that nodes of a Decentralized Oracle Network (DON) interact with
+ * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
+ */
+contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+  uint16 constant REQUEST_DATA_VERSION = Functions.REQUEST_DATA_VERSION;
+
+  event OracleRequest(
+    bytes32 indexed requestId,
+    address requestingContract,
+    address requestInitiator,
+    uint64 subscriptionId,
+    address subscriptionOwner,
+    bytes data
+  );
+  event OracleResponse(bytes32 indexed requestId);
+  event UserCallbackError(bytes32 indexed requestId, string reason);
```

Comment:
I don't think we need this for the node (the listener).

Reply:
For the listener, it seems we only need `OracleResponse` event if you guarantee it will be emitted regardless of the outcome (e.g. user error or not, the listener does not care).
But @bolekk can add more input here.

Reply:
Yes, the node only cares if the request was processed in the contract.

Reply:
Removed this and `ResponseTransmitted`